### PR TITLE
Simplify setting up of local TTS Dev environment

### PR DESCRIPTION
**Pull request by KrishnaIyer**
#### Summary

Simplify setting up of local TTS Dev environment

#### Changes

- Add clear instructions on local Dev env setup.
- Separate configuration for docker and non-docker instance of the stack.

#### Testing

Running the instructions locally.

##### Regressions

Tested the `Getting Started` guide locally to make sure that still works.

#### Notes for Reviewers

Please read my reasoning for this before commenting
- This is meant to be the simplest guide to get someone started with a working dev environment (It involves running 4 commands).
- I'm splitting the config files since we don't want to create a `/run/secrets` locally.
- I'm leaving TLS out of this (since 99.99% of local dev does not need TLS). I'll add a small note on it or leave it out entirely if necessary.
- `ttn-lw-stack.yml` currently contains nothing (except the certs though we don't use them). I don't want to add lots of config here and confuse first time users.

@laurensslats / @nejraselimovic : Please run this and let me know how your UX is. I'll create a TTES version after this.

#### Checklist

- [ ] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md`. The target branch is set to `master` if the changes are fully compatible with existing API, database, configuration and CLI.
- [x] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -25,14 +25,60 @@ $ make init
...
+## Running a development build of The Things Stack
+
+This section explains how to get a bare-bones version of The Things Stack running on your local machine. This will build whatever code is present in your local repository (along with local changes) and run in it using the default ports.
+
+If you want to just run a docker image of The Things Stack, then check the [Installation](https://thethingsstack.io/latest/getting-started/installation/) section of the documentation.
+
+> Note: Make sure that you've run `$ make init` before following this section.
+
+### Pre-requisites
+This section requires [`docker`](https://www.docker.com/get-started) to be installed.
```
Comment (htdvisser) [inline DEVELOPMENT.md:37]: Empty line between blocks (h3 and p)

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -25,14 +25,60 @@ $ make init
...
-This starts a CockroachDB and Redis database in Docker containers, creates a database, migrates tables and creates a user `admin` with password `admin`.
+This creates a database, migrates tables and creates a user `admin` with password `admin`.
+
+4. Start an development instance of The Things Stack
+
+```bash
+$ go run ./cmd/ttn-lw-stack -c ./config/stack/ttn-lw-stack.yml start
+```
+
+5. Login to the stack via the Console
```
Comment (htdvisser) [inline DEVELOPMENT.md:75]: We don't call it "the stack"

File: DEVELOPMENT.md [hunk 3]
```diff
@@ -25,14 +25,60 @@ $ make init
...
+
+In a web browser, navigate to `http://localhost:1885/` and login using credentials from step 3.
+
+6. Customizing configuration
+
+To customize the configuration, edit the `/config/stack/ttn-lw-stack.yml` file. The configuration is documented in the [Configuration Reference](https://thethingsstack.io/latest/reference/configuration/).
```
Comment (htdvisser) [inline DEVELOPMENT.md:81]: Maybe better to recommend creating a copy and pointing to that?

Comment (KrishnaIyer) [reply DEVELOPMENT.md:81]: Indeed. Actually my original idea was that the script would make a copy of the config in `.env` and that's freely editable.

**Comment by benolayinka** [comment]
This is great. On my first go it didn't work because I still had the environment variables loaded from [development configuration](https://github.com/TheThingsNetwork/lorawan-stack/blob/f84629d0457a33250f9d8435f7ccadf7b7c364f3/DEVELOPMENT.md#development-configuration), so maybe worth mentioning in troubleshooting that this can happen if you've previously been using `./mage js:serve` to hot reload the console.

**Comment by KrishnaIyer** [comment]
Ok I've made all the small changes. Waiting on @laurensslats to check if this works as intended.

**Comment by laurensslats** [comment]
When running the command `$ ./mage js:build` the following error is received:
```
Error: `Cannot find module 'es-abstract/2019/CreateDataProperty'` 
```

Thoughts? @benolayinka @KrishnaIyer 


Full error message:
```
$ ./mage js:build
/Users/Laurens/go/src/go.thethings.network/lorawan-stack/node_modules/webpack-cli/bin/cli.js:93
				throw err;
				^

Error: Cannot find module 'es-abstract/2019/CreateDataProperty'
Require stack:
- /Users/Laurens/go/src/go.thethings.network/lorawan-stack/node_modules/object.getownpropertydescriptors/implementation.js
- /Users/Laurens/go/src/go.thethings.network/lorawan-stack/node_modules/object.getownpropertydescriptors/index.js
- /Users/Laurens/go/src/go.thethings.network/lorawan-stack/node_modules/util.promisify/implementation.js
- /Users/Laurens/go/src/go.thethings.network/lorawan-stack/node_modules/util.promisify/index.js
- /Users/Laurens/go/src/go.thethings.network/lorawan-stack/node_modules/html-webpack-plugin/index.js
- /Users/Laurens/go/src/go.thethings.network/lorawan-stack/config/webpack.config.babel.js
- /Users/Laurens/go/src/go.thethings.network/lorawan-stack/node_modules/webpack-cli/bin/utils/convert-argv.js
- /Users/Laurens/go/src/go.thethings.network/lorawan-stack/node_modules/webpack-cli/bin/cli.js
- /Users/Laurens/go/src/go.thethings.network/lorawan-stack/node_modules/webpack/bin/webpack.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:957:15)
    at Function.Module._load (internal/modules/cjs/loader.js:840:27)
    at Module.require (internal/modules/cjs/loader.js:1019:19)
    at require (/Users/Laurens/go/src/go.thethings.network/lorawan-stack/node_modules/v8-compile-cache/v8-compile-cache.js:161:20)
    at Object.<anonymous> (/Users/Laurens/go/src/go.thethings.network/lorawan-stack/node_modules/object.getownpropertydescriptors/implementation.js:3:26)
    at Module._compile (/Users/Laurens/go/src/go.thethings.network/lorawan-stack/node_modules/v8-compile-cache/v8-compile-cache.js:192:30)
    at Module._compile (/Users/Laurens/go/src/go.thethings.network/lorawan-stack/node_modules/pirates/lib/index.js:99:24)
    at Module._extensions..js (internal/modules/cjs/loader.js:1153:10)
    at Object.newLoader [as .js] (/Users/Laurens/go/src/go.thethings.network/lorawan-stack/node_modules/pirates/lib/index.js:104:7)
    at Module.load (internal/modules/cjs/loader.js:977:32)
    at Function.Module._load (internal/modules/cjs/loader.js:877:14)
    at Module.require (internal/modules/cjs/loader.js:1019:19)
    at require (/Users/Laurens/go/src/go.thethings.network/lorawan-stack/node_modules/v8-compile-cache/v8-compile-cache.js:161:20)
    at Object.<anonymous> (/Users/Laurens/go/src/go.thethings.network/lorawan-stack/node_modules/object.getownpropertydescriptors/index.js:5:22)
    at Module._compile (/Users/Laurens/go/src/go.thethings.network/lorawan-stack/node_modules/v8-compile-cache/v8-compile-cache.js:192:30)
    at Module._compile (/Users/Laurens/go/src/go.thethings.network/lorawan-stack/node_modules/pirates/lib/index.js:99:24)
    at Module._extensions..js (internal/modules/cjs/loader.js:1153:10)
    at Object.newLoader [as .js] (/Users/Laurens/go/src/go.thethings.network/lorawan-stack/node_modules/pirates/lib/index.js:104:7)
    at Module.load (internal/modules/cjs/loader.js:977:32)
    at Function.Module._load (internal/modules/cjs/loader.js:877:14)
    at Module.require (internal/modules/cjs/loader.js:1019:19)
    at require (/Users/Laurens/go/src/go.thethings.network/lorawan-stack/node_modules/v8-compile-cache/v8-compile-cache.js:161:20)
    at Object.<anonymous> (/Users/Laurens/go/src/go.thethings.network/lorawan-stack/node_modules/util.promisify/implementation.js:12:33)
    at Module._compile (/Users/Laurens/go/src/go.thethings.network/lorawan-stack/node_modules/v8-compile-cache/v8-compile-cache.js:192:30)
    at Module._compile (/Users/Laurens/go/src/go.thethings.network/lorawan-stack/node_modules/pirates/lib/index.js:99:24)
    at Module._extensions..js (internal/modules/cjs/loader.js:1153:10)
    at Object.newLoader [as .js] (/Users/Laurens/go/src/go.thethings.network/lorawan-stack/node_modules/pirates/lib/index.js:104:7)
    at Module.load (internal/modules/cjs/loader.js:977:32)
    at Function.Module._load (internal/modules/cjs/loader.js:877:14)
    at Module.require (internal/modules/cjs/loader.js:1019:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/Laurens/go/src/go.thethings.network/lorawan-stack/node_modules/object.getownpropertydescriptors/implementation.js',
    '/Users/Laurens/go/src/go.thethings.network/lorawan-stack/node_modules/object.getownpropertydescriptors/index.js',
    '/Users/Laurens/go/src/go.thethings.network/lorawan-stack/node_modules/util.promisify/implementation.js',
    '/Users/Laurens/go/src/go.thethings.network/lorawan-stack/node_modules/util.promisify/index.js',
    '/Users/Laurens/go/src/go.thethings.network/lorawan-stack/node_modules/html-webpack-plugin/index.js',
    '/Users/Laurens/go/src/go.thethings.network/lorawan-stack/config/webpack.config.babel.js',
    '/Users/Laurens/go/src/go.thethings.network/lorawan-stack/node_modules/webpack-cli/bin/utils/convert-argv.js',
    '/Users/Laurens/go/src/go.thethings.network/lorawan-stack/node_modules/webpack-cli/bin/cli.js',
    '/Users/Laurens/go/src/go.thethings.network/lorawan-stack/node_modules/webpack/bin/webpack.js'
  ]
}
Error: running "node_modules/.bin/webpack --config config/webpack.config.babel.js" failed with exit code 1
```

**Comment by KrishnaIyer** [comment]
@laurensslats : Have you first setup the [development environment](https://github.com/TheThingsNetwork/lorawan-stack/blob/99e48c1d2170322a6eb1f8e0ba9ea2c18907cdaa/DEVELOPMENT.md#development-environment)?

**Comment by laurensslats** [comment]
Error received when running: `./mage js:build`

```
Error: running "node_modules/.bin/webpack --config config/webpack.config.babel.js" failed with exit code 2
```

When neglecting this error and continue following the guide, I get a blank screen: http://localhost:1885/console/

Screenshot 2020-06-16 at 16 06 53

@benolayinka did you face any similar issue?


Console logs:
```
  INFO Client error                             duration=30.612µs method=GET namespace=web remote_addr=[::1]:53502 request_id=01EAYQ1MR74P3QP5YB58DR4SCJ response_size=19 status=404 url=http://localhost:1885/assets/console.css
  INFO Client error                             duration=35.414µs method=GET namespace=web remote_addr=[::1]:53503 request_id=01EAYQ1MR8BFJ5V8Z5WXQV1WRH response_size=19 status=404 url=http://localhost:1885/assets/console.js
  INFO Client error                             duration=79.458µs method=GET namespace=web remote_addr=[::1]:53503 request_id=01EAYQ1N9WD1DG3QCV91PHQ6ME response_size=19 status=404 url=http://localhost:1885/assets/console.css
  INFO Client error                             duration=42.783µs method=GET namespace=web remote_addr=[::1]:53503 request_id=01EAYQ1NA7DN322CNH16BVW3RF response_size=19 status=404 url=http://localhost:1885/assets/console.js
  INFO Request handled                          duration=11.246355ms method=GET namespace=web remote_addr=[::1]:53503 request_id=01EAYQ38W2P5Y1M0C245F9H05Q response_size=754 status=200 url=http://localhost:1885/console/
  INFO Client error                             duration=25.781µs method=GET namespace=web remote_addr=[::1]:53503 request_id=01EAYQ39103YAB20YPE9DZVE0C response_size=19 status=404 url=http://localhost:1885/assets/console.css
  INFO Client error                             duration=29.09µs method=GET namespace=web remote_addr=[::1]:53503 request_id=01EAYQ39186KRR609P632MQJ4E response_size=19 status=404 url=http://localhost:1885/assets/console.js
  INFO Client error                             duration=44.798µs method=GET namespace=web remote_addr=[::1]:53503 request_id=01EAYQ397WFBDE011H67Y22CZJ response_size=19 status=404 url=http://localhost:1885/assets/console.js
  INFO Client error                             duration=34.489µs method=GET namespace=web remote_addr=[::1]:53503 request_id=01EAYQ39FBZ251R7C39569V1N8 response_size=19 status=404 url=http://localhost:1885/assets/console-favicon.svg
  INFO Client error                             duration=28.186µs method=GET namespace=web remote_addr=[::1]:53503 request_id=01EAYQ39FTZ3PEMPRGTEN49XZJ response_size=19 status=404 url=http://localhost:1885/assets/console-favicon.png
```

**Comment by KrishnaIyer** [comment]
Awesome. @laurensslats: Let's discuss your issues in a thread.

**Pull request merged**